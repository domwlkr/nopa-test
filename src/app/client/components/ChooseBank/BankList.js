import React, { PropTypes } from 'react';
import BankItem from './BankItem';

const BankList = (props) => {
  const banks = props.bankList.map((bank, index) => (
    <BankItem isSelected={bank.name == props.selectedBank} key={index} bank={bank} onSelectBank={props.onSelectBank} />
  ));

  return (
    <ul className="bank-list">
      {banks}
    </ul>
  );
};

BankList.propTypes = {
  bankList: PropTypes.array.isRequired,
  onSelectBank: PropTypes.func.isRequired,
  selectedBank: PropTypes.object.isRequired
};

export default BankList;