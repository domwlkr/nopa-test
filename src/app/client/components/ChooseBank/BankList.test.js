import React from 'react';
import BankList from './BankList';
import initialState from '../../redux/reducers/initialState.js';
import renderer from 'react-test-renderer';

const bankList = initialState.bankList;
const onSelectBank = jest.fn();
const selectedBank = {
  name: 'Barclays',
  logo: 'test-file-stub'
};

test('Renders correctly', () => {
  const tree = renderer.create(
    <BankList bankList={bankList} onSelectBank={onSelectBank} selectedBank={selectedBank} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});