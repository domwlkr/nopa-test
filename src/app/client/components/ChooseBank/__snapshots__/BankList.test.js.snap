// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<ul
  className="bank-list"
>
  <li
    className="bank-item "
    onClick={[Function]}
  >
    <img
      alt="Barclays"
      className="bank-image"
      src="test-file-stub"
    />
  </li>
  <li
    className="bank-item "
    onClick={[Function]}
  >
    <img
      alt="Natwest"
      className="bank-image"
      src="test-file-stub"
    />
  </li>
  <li
    className="bank-item "
    onClick={[Function]}
  >
    <img
      alt="Lloyds"
      className="bank-image"
      src="test-file-stub"
    />
  </li>
  <li
    className="bank-item "
    onClick={[Function]}
  >
    <img
      alt="HSBC"
      className="bank-image"
      src="test-file-stub"
    />
  </li>
  <li
    className="bank-item "
    onClick={[Function]}
  >
    <img
      alt="TSB"
      className="bank-image"
      src="test-file-stub"
    />
  </li>
  <li
    className="bank-item "
    onClick={[Function]}
  >
    <img
      alt="Santander"
      className="bank-image"
      src="test-file-stub"
    />
  </li>
</ul>
`;
